cpu/esp32/Makefile.include

```diff
@@ -126,7 +126,9 @@ ifneq (,$(filter esp_wifi_any,$(USEMODULE)))
     LINKFLAGS += $(BINDIR)/esp_idf_esp32.a
     LINKFLAGS += $(BINDIR)/esp_idf_nvs_flash.a
     LINKFLAGS += $(BINDIR)/esp_idf_spi_flash.a
+    LINKFLAGS += $(BINDIR)/pthread.a
```

Comment:
Putting the libraries into the linker command should already be done by https://github.com/RIOT-OS/RIOT/blob/98f42eeea8e04b0e2109aa77e04d51f566f60bee/makefiles/modules.inc.mk#L7

However I see from the other parts that you duplicate dependency handling in both in `Makefile.include` and `Makefile.dep`. This comes because dependencies are resolved after board and cpu `Makefile.include` and I am currently trying to address it https://github.com/RIOT-OS/RIOT/issues/9913

Reply:
@cladmi 
> Putting the libraries into the linker command should already be done by

Unfortunatly, it is a bit more tricky and I was not able to figure out another solution when I had to add the vendor libraries:
```
 LINKFLAGS += -lcore -lrtc -lnet80211 -lpp -lsmartconfig -lcoexist
 LINKFLAGS += -lwps -lwpa -lwpa2 -lespnow -lmesh -lphy -lstdc++
```
References in these libraries could only be resolved only by grouping them with module archives
```
esp_idf.a
esp_idf_esp32.a
esp_idf_nvs_flash.a
esp_idf_spi_flash.a
riot_freertos.a
```
even though they were already added by RIOT module makefiles. But they were in another ```--start-group ... --end-group``` and could not figure out how to add further libraries to this list. Furthermore, these module archives reference symbols from other RIOT modules. So I had also to group
```
xtimer.a
core.a
pthread.a
```
with the libraries to be able to resolve all symbols.

With your hint to the BASELIBS variable, I could solve the problem as following:
```
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libcore.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/librtc.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libnet80211.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libpp.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libsmartconfig.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libcoexist.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libwps.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libwpa.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libwpa2.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libespnow.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libmesh.a
BASELIBS += $(ESP32_SDK_DIR)/components/esp32/lib/libphy.a
```
Would this be the right way?

Reply:
@cladmi 
> However I see from the other parts that you duplicate dependency handling in both in Makefile.include and Makefile.dep.

Yes, the reason was that the dependencies are included too late.

Reply:
With this diff to your PR it also compiles for your test command (not tested!)

```
diff --git a/cpu/esp32/Makefile.include b/cpu/esp32/Makefile.include
index 12ff0b1af..89e3d3814 100644
--- a/cpu/esp32/Makefile.include
+++ b/cpu/esp32/Makefile.include
@@ -121,21 +121,8 @@ LINKFLAGS += -L$(ESP32_SDK_DIR)/components/esp32/lib
 LINKFLAGS += -Wl,--start-group

 ifneq (,$(filter esp_wifi_any,$(USEMODULE)))
-    LINKFLAGS += $(BINDIR)/cpu.a
-    LINKFLAGS += $(BINDIR)/esp_idf.a
-    LINKFLAGS += $(BINDIR)/esp_idf_esp32.a
-    LINKFLAGS += $(BINDIR)/esp_idf_nvs_flash.a
-    LINKFLAGS += $(BINDIR)/esp_idf_spi_flash.a
-    LINKFLAGS += $(BINDIR)/pthread.a
-    LINKFLAGS += $(BINDIR)/riot_freertos.a
-    LINKFLAGS += $(BINDIR)/xtimer.a
-    LINKFLAGS += -lcore -lrtc -lnet80211 -lpp -lsmartconfig -lcoexist
-    LINKFLAGS += -lwps -lwpa -lwpa2 -lespnow -lmesh -lphy -lstdc++
-endif
-
-ifneq (,$(filter pthread,$(USEMODULE)))
-    LINKFLAGS += $(BINDIR)/core.a
-    LINKFLAGS += $(BINDIR)/pthread.a
+    BASELIBS += -lcore -lrtc -lnet80211 -lpp -lsmartconfig -lcoexist
+    BASELIBS += -lwps -lwpa -lwpa2 -lespnow -lmesh -lphy -lstdc++
 endif

 LINKFLAGS += -lhal -lg -lc -lg
```

When putting `-lhal -lg -lc -lg` into `BASELIBS` it failed to compiled but did not investigate further.

Reply:
@cladmi Thanks, based on your hints I was able to further reduce the special stuff.

Reply:
Cool :) Could you split this part out as it can be merged alone.

The handling of `INCLUDES` path and conflicts is a different topic where there are other things to consider. I am also still not sure of what to do there.

Reply:
@cladmi Do you mean as a separate pull request or just a separate commit?

Reply:
I would say separate PRs as I could merge the `BASELIBS` changes alone earlier.

Reply:
I wanted to have the split merged before this one but I forgot to put the label >< My bad.